tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -0,0 +1,202 @@
+                     instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableUpDownCounter`1"))
+            {
+                aggregationType = "UpDownCounter";
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Gauge`1") ||
+                     instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableGauge`1"))
+            {
+                aggregationType = "Gauge";
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Histogram`1"))
+            {
+                aggregationType = "Histogram";
+            }
+            else
+            {
+                Log.Debug("Skipping unsupported instrument: {InstrumentName} of type: {InstrumentType}", instrument.Name, instrumentType);
+                return;
+            }
+
+            var doubleValue = Convert.ToDouble(value);
+
+            // RFC requirement: Validate negative values for Counter and Histogram
+            if ((aggregationType == "Counter" || aggregationType == "Histogram") && doubleValue < 0)
+            {
+                Log.Warning("Ignoring negative value {Value} for {InstrumentType} instrument: {InstrumentName}. API usage is incorrect.", doubleValue, aggregationType, instrument.Name);
+                return;
+            }
+
+            var meterName = instrument.Meter.Name;
+            var key = string.Concat(meterName, ".", instrument.Name);
+            var temporality = GetTemporalityString();
+
+            var tagsDict = tags.Length == 0 ? EmptyTags : CreateTagsDictionary(tags);
+
+            CapturedMetrics.AddOrUpdate(
+                key,
+                _ =>
+                {
+                    var newMetric = new MetricPoint(instrument.Name, meterName, aggregationType, temporality, tagsDict, isInteger);
+                    UpdateMetricPoint(newMetric, aggregationType, doubleValue);
+                    return newMetric;
+                },
+                (_, existing) =>
+                {
+                    existing.IsIntegerValue = isInteger;
+                    UpdateMetricPoint(existing, aggregationType, doubleValue);
+                    return existing;
+                });
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void UpdateMetricPoint(MetricPoint metricPoint, string aggregationType, double value)
+        {
+            switch (aggregationType)
+            {
+                case "Counter":
+                case "UpDownCounter":
+                    metricPoint.UpdateCounter(value);
+                    break;
+                case "Gauge":
+                    metricPoint.UpdateGauge(value);
+                    break;
+                case "Histogram":
+                    metricPoint.UpdateHistogram(value);
+                    break;
+            }
+        }
+
+        internal static void TriggerAsyncCollection()
```

Comment:
I don't think `TriggerAsyncCollection` is used in this PR or the next one, since you call `MetricReader.CollectObservableInstruments` in your test code. You can delete this